b2luigi/batch/processes/htcondor.py

```diff
@@ -0,0 +1,225 @@
+import json
+import logging
+import os
+import re
+import shutil
+import stat
+import subprocess
+import enum
+import sys
+
+from b2luigi.core.utils import get_setting, get_output_dirs
+from b2luigi.batch.processes import BatchProcess, JobStatus, BatchJobStatusCache
+from b2luigi.core.utils import get_log_file_dir
+
+
+class HTCondorJobStatusCache(BatchJobStatusCache):
+
+    def __init__(self):
+        super(HTCondorJobStatusCache, self).__init__()
+
+    def _ask_for_job_status(self, job_id: int = None):
+        q_cmd = ["condor_q", "-json", "-attributes", "ClusterId,ProcId,JobStatus"]
+        history_cmd = ["condor_history", "-json", "-attributes", "ClusterId,ProcId,JobStatus"]
+
+        if job_id:
+            q_cmd.append(str(job_id))
+            output = subprocess.check_output(q_cmd)
+        else:
+            output = subprocess.check_output(q_cmd)
+        output = output.decode()
+
+        dict_list = []
+        for status_dict in output.lstrip("[\n").rstrip("\n]\n").split("\n,\n"):
+            if status_dict:
+                dict_list.append(json.loads(status_dict))
+            else:
+                continue
+
+        if job_id:
+            if job_id not in [status_dict["ClusterId"] for status_dict in dict_list]:
+                history_cmd.append(str(job_id))
+                output = subprocess.check_output(history_cmd)
+                output = output.decode()
+                output = output.lstrip("[\n").rstrip("\n]\n").split("\n,\n")
+                if output:
+                    for status_dict in output:
+                        dict_list.append(json.loads(status_dict))
+
+        for status_dict in dict_list:
+            self[status_dict["ClusterId"]] = status_dict["JobStatus"]  # int -> int
+
+
+class HTCondorJobStatus(enum.IntEnum):
+    idle = 1
+    running = 2
+    removed = 3
+    completed = 4
+    held = 5
+    submission_err = 6
+
+
+_batch_job_status_cache = HTCondorJobStatusCache()
+
+
+class HTCondorProcess(BatchProcess):
+    """
+    Reference implementation of the batch process for a HTCondor batch system.
+
+    We assume that the batch system shares a file system with the submission node you
+    are currently working on (or at least the current folder and the result/log directory
+    are also available there with the same path).
+    The environment has to be set up on its own new for each job. For that, a environment
+    setup script has to be provided in the ``settings.json`` file.
```

Comment:
Can I also have this as a task-specific setting?
Can I also just give you a list of env variables (as in the LSF task case)?

Reply:
The documentation states that we could also specify environment variables in the job description file. I also found a `getenv` option which should copy all current variables. I have to check again if in some cases one can ommit setting up a new environment.